### ISSUE TITLE
Only reporting one browser worth of tests (mocha)

### ISSUE THREAD

**Issue by treyturner** [webdriverio/wdio-allure-reporter#52]
Hi,
I'm using `webdriverio 4.9.11` with `wdio-mocha-framework 0.5.11` and `wdio-allure-reporter 0.1.2`. In Jenkins, I run all my tests with four browsers defined in `wdio.conf.js`. I'm finding that although it appears all the appropriate .json/.xml/.png files are created in `allure-results/`, when I generate the report, only one set of test results appear instead of four. Each run, an arbitrary browser is listed in the report under the Environment section. Furthermore, it appears test results are being overwritten; if I have a test fail on one browser, the passing tests from other browsers can overwrite it in the report and show the test as passed, even if it failed using the same browser listed in the Environment section. I have 35 tests at present, so when I run against four browsers, I expect to see 140 test results. Instead I only see 35.

I'm wondering if the reporter needs to output environment information differently, or prepend the browser name onto test names so they don't overwrite each other. I'm also open to the idea that I've missed a configuration step or have done something else wrong, but I've scoured the docs and asked in gitter but haven't made any progress so far.

Here's a minimal reproducible example, just run `npm install && npm test`: https://github.com/treyturner/wdio-allure-multibrowser-test

Thanks for your time. as well as your work on this reporter and the project as a whole.

**Comment by BorisOsipov** [comment]
@treyturner In your case Allure didn't overwrite test results. 
Your tests results handle as retries of the same test due equal tests names.
See [screenshot](https://content.screencast.com/users/Boris_Osipov/folders/Jing/media/daaedc50-5554-4966-bfa8-d7607799ac72/2018-01-03_0112.png)

**Comment by treyturner** [comment]
An interesting note for sure. I still require to separate and surface the various test results of browser/OS permutations.

**Comment by netmajor** [comment]
Hi, I have also check this scenario. I have executed same tests but using two browsers. After tests I have in .allure-result two files per browser, but allure generate report with only information for one test.  I ask allure support  what option should contains test that report show statistics for test per browser. I checked feature and environment but its not works.

For me solving this problem is also important because in project I am using wdio-allure-reporter I have in plans to execute test for different browsers.

I am not sure how I want to see that information on the report...
